common/djangoapps/third_party_auth/api/views.py

```diff
@@ -89,3 +97,136 @@ def get(self, request, username):
         return Response({
             "active": active_providers
         })
+
+
+class UserMappingView(ListAPIView):
+    """
+    Map between the third party auth account IDs (remote_id) and EdX username.
+
+    This API is intended to be a server-to-server endpoint. An on-campus middleware or system should consume this.
+
+    **Use Case**
+
+        Get a paginated list of mappings between edX users and remote user IDs for all users currently
+        linked to the given backend.
+
+        The list can be filtered by edx username or third party ids. The filter is limited by the max length of URL.
+        It is suggested to query no more than 50 usernames or remote_ids in each request to stay within above
+        limitation
+
+        Each page in the list can contain up to 50 mappings.
```

Comment:
It is actually by page_size. I've updated the docstring.

On Tue, Oct 27, 2015 at 2:56 PM Nimisha Asthagiri <notifications@github.com>
wrote:

> In common/djangoapps/third_party_auth/api/views.py
> <https://github.com/edx/edx-platform/pull/9842#discussion_r43193347>:
>
> > +class UserMappingView(ListAPIView):
> > +    """
> > +    Map between the third party auth account IDs (remote_id) and EdX username.
> > +
> > +    This API is intended to be a server-to-server endpoint. An on-campus middleware or system should consume this.
> > +
> > +    **Use Case**
> > +
> > +        Get a paginated list of mappings between edX users and remote user IDs for all users currently
> > +        linked to the given backend.
> > +
> > +        The list can be filtered by edx username or third party ids. The filter is limited by the max length of URL.
> > +        It is suggested to query no more than 50 usernames or remote_ids in each request to stay within above
> > +        limitation
> > +
> > +        Each page in the list can contain up to 50 mappings.
>
> Isn't this governed by the page_size? Or is 50 a hard limit?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/edx/edx-platform/pull/9842/files#r43193347>.
>